swift/storage.py

```diff
@@ -168,37 +172,46 @@ def __init__(self, **settings):
         }
         self.os_options.update(self.os_extra_options)
 
-        # Get authentication token
-        self.storage_url, self.token = swiftclient.get_auth(
-            self.api_auth_url,
-            self.api_username,
-            self.api_key,
-            auth_version=self.auth_version,
-            os_options=self.os_options)
-        self.http_conn = swiftclient.http_connection(self.storage_url)
+        if self.auto_base_url:
+            # Will be set when storage_url is set
+            self.base_url = None
+        else:
+            self.base_url = self.override_base_url
 
-        # Check container
-        try:
-            swiftclient.head_container(self.storage_url,
-                                       self.token,
-                                       self.container_name,
-                                       http_conn=self.http_conn)
-        except swiftclient.ClientException:
-            headers = {}
-            if self.auto_create_container:
-                if self.auto_create_container_public:
-                    headers['X-Container-Read'] = '.r:*'
-                if self.auto_create_container_allow_orgin:
-                    headers['X-Container-Meta-Access-Control-Allow-Origin'] = \
-                        self.auto_create_container_allow_orgin
-                swiftclient.put_container(self.storage_url,
-                                          self.token,
-                                          self.container_name,
-                                          http_conn=self.http_conn,
-                                          headers=headers)
-            else:
-                raise ImproperlyConfigured(
-                    "Container %s does not exist." % self.container_name)
+        self.http_conn = None
+        if not self.lazy_connect:
+            self.get_token()
+
+    def get_token(self):
+        if time() - self._token_creation_time >= self.auth_token_duration:
+            self.storage_url, new_token = swiftclient.get_auth(
```

Comment:
Why not simply `self.storage_url, self.toke = ...`?

Reply:
Ah good point -- I think it's just leftover from the previous code.

Fixed with d8f3352